src/tests/resistance/handshake.rs

```diff
@@ -101,6 +111,137 @@ async fn r001_t2_HANDSHAKE_reject_if_server_too_long() {
     node.stop().unwrap();
 }
 
+/// Decide whether to enable node logs and tracing for synthetic nodes.
+#[derive(Clone, Copy)]
+enum Debug {
+    On,
+    Off,
+}
+
+impl Debug {
+    // This API exists just so we could enable the synth tracing once,
+    // because calling that function twice would break the test.
+    fn enable() -> Self {
+        synth_node::enable_tracing();
+        Self::On
+    }
+
+    fn disable() -> Self {
+        // We should use something like synth_node::disable_tracing here (still unimplemented),
+        // but we'll never use it anyway so this is good enough
+        Self::Off
+    }
+
+    /// Convert to boolean value.
+    fn is_on(self) -> bool {
+        match self {
+            Self::On => true,
+            Self::Off => false,
+        }
+    }
+}
+
+// Runs the handshake request test with a given handshake configuration.
+// Returns the truthful fact about the relationship with the node.
+async fn run_handshake_req_test_with_cfg(cfg: SynthNodeCfg, debug: Debug) -> bool {
+    // Spin up a node instance.
+    let target = TempDir::new().expect(ERR_TEMPDIR_NEW);
+    let mut node = Node::builder()
+        .log_to_stdout(debug.is_on())
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .expect(ERR_NODE_BUILD);
+
+    // Create a synthetic node and enable handshaking.
+    let mut synthetic_node = SyntheticNode::new(&cfg).await;
+
+    // Connect to the node and initiate the handshake.
+    let handshake_established = if synthetic_node.connect(node.addr()).await.is_err() {
+        false
+    } else {
+        // Wait for any message.
+        synthetic_node
+            .expect_message(&|m: &BinaryMessage| matches!(&m, _))
+            .await
+    };
+
+    if debug.is_on() {
```

Comment:
no need to wait in case of successful handshake
```suggestion
    if debug.is_on() && !handshake_established {
```

Reply:
Done